src/core/server/saved_objects/migrations/core/index_migrator.ts

```diff
@@ -143,7 +143,9 @@ async function deleteIndexTemplates({ client, log, obsoleteIndexTemplatePattern
     return;
   }
 
-  const { body: templates } = await client.cat.templates<Array<{ name: string }>>({
+  const {
+    body: { records: templates },
```

Comment:
I'm not sure we should be using this `records` key in the response or not. Needs to be tested here in other usages of the cat API.